pkg/kubernetes/reconcile.go

```diff
@@ -113,6 +113,11 @@ func Reconcile(raw map[string]interface{}, spec v1alpha1.Spec, targets []*regexp
 			return out[i].Kind() < out[j].Kind()
 		}
 
+		// If namespaces differ, sort by the namespace.
+		if out[i].Metadata().Namespace() != out[j].Metadata().Namespace() {
```

Comment:
Does is consider `"" == spec.namespace`?

Reply:
IIUC empty string is not a valid namespace; and is also returned by the Namespace() function if namespace is not set. For the purposes of sorting, this defines an order over namespaces if they are not equal, so we should be fine here.